### PR TITLE
keras to torch export

### PR DESCRIPTION
**Pull request by pctablet505**
This PR adds torch export path for Keras models using `model.export(..., format="torch")`. It is split out from #22758 so the `.pt2` export support can be reviewed independently from LiteRT conversion.

### What this adds

- Adds `keras.src.export.export_torch` backed by `torch.export.export` and `torch.export.save`.
- Exposes `format="torch"` from `Model.export()` with `.pt2` filepath validation and PyTorch-backend gating.
- Forwards supported `torch.export.export` kwargs, including `strict`, `dynamic_shapes`, `preserve_module_call_signature`, and `prefer_deferred_runtime_asserts_over_guards`.
- Defaults `strict=False` for broader compatibility with Keras models.
- Normalizes single-input `dynamic_shapes` specs for Keras models, whose Torch module signature is `*args, **kwargs`.
- Includes backend Torch fixes needed by real exported models: slice/slice_update tracing, contiguous channels-last transposes, and static-shape-preserving scalar repeat.

### Export flow

```mermaid
flowchart TD
    A[model.export format torch] --> B[Validate .pt2 filepath]
    B --> C[Infer or use input_signature]
    C --> D[Create torch sample inputs]
    D --> E[Normalize torch.export kwargs]
    E --> F[torch.export.export strict false by default]
    F --> G[torch.export.save]
    G --> H[Load later with torch.export.load]
    H --> I[Run ExportedProgram.module]
```

### Tests

- `KERAS_BACKEND=torch PYTHONPATH=/usr/local/google/home/hellorahul/projects/keras /usr/local/google/home/hellorahul/projects/venv313/bin/python -m pytest /usr/local/google/home/hellorahul/projects/keras/keras/src/export/torch_test.py -q --tb=short`
- `KERAS_BACKEND=torch PYTHONPATH=/usr/local/google/home/hellorahul/projects/keras /usr/local/google/home/hellorahul/projects/venv313/bin/python -m pytest /usr/local/google/home/hellorahul/projects/keras/keras/src/ops/core_test.py -q -k "slice" --tb=short`
- `KERAS_BACKEND=torch PYTHONPATH=/usr/local/google/home/hellorahul/projects/keras /usr/local/google/home/hellorahul/projects/venv313/bin/python -m pytest /usr/local/google/home/hellorahul/projects/keras/keras/src/ops/numpy_test.py -q -k "repeat" --tb=short`
- `KERAS_BACKEND=torch PYTHONPATH=/usr/local/google/home/hellorahul/projects/keras /usr/local/google/home/hellorahul/projects/venv313/bin/python -m pytest /usr/local/google/home/hellorahul/projects/keras/keras/src/layers/pooling/adaptive_pooling1d_test.py /usr/local/google/home/hellorahul/projects/keras/keras/src/layers/pooling/adaptive_pooling2d_test.py /usr/local/google/home/hellorahul/projects/keras/keras/src/layers/pooling/adaptive_pooling3d_test.py -q --tb=short`
- Clean temporary worktree: `/usr/local/google/home/hellorahul/.local/bin/pre-commit run --all-files`

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/export/torch.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            return [(dynamic_shapes,)]
+        return dynamic_shapes
+
+    if isinstance(dynamic_shapes, (list, tuple)):
+        if len(dynamic_shapes) == 1 and isinstance(
+            dynamic_shapes[0], (list, tuple)
+        ):
+            return dynamic_shapes
+        if len(dynamic_shapes) == len(sample_inputs):
+            return [tuple(dynamic_shapes)]
```
[the remark this answers is not in the document]
Comment (pctablet505) [reply keras/src/export/torch.py:169]: Keras models use forward(*args, **kwargs), so torch.export treats the entire args tuple as a single positional argument. For a single-input model, args = (tensor,), so dynamic_shapes must be [({0: batch},)] — a list containing the spec for the args tuple.

The bare (dynamic_shapes,) format only works for modules with explicit forward(self, x) signatures, not Keras's *args.

File: keras/src/backend/torch/core.py [hunk 2]
```diff
@@ -617,32 +617,56 @@ def scatter_update(inputs, indices, updates, reduction=None):
...
+
     slices = [
-        python_slice(start_index, start_index + update_length)
+        builtins.slice(start_index, start_index + update_length)
         for start_index, update_length in zip(start_indices, updates.shape)
     ]
     outputs = torch.clone(inputs)
-    outputs[slices] = updates
+    outputs[tuple(slices)] = updates
     return outputs
```
[the remark this answers is not in the document]
Comment (pctablet505) [reply keras/src/backend/torch/core.py]: the suggested `torch.narrow + .copy_()` approach was tested but fails with `PendingUnbackedSymbolNotFound` because `start_indices[dim]` creates unbacked symbols that don't appear in narrow's output shape.

Instead, the fix uses `torch.arange + meshgrid + index_put_`, which is fully traceable by `torch.export` and avoids extracting scalar values from symbolic tensors. A new test `test_export_with_dynamic_slice_update` was added to verify this path.

**Review by hertschuh — APPROVED** [review]
Thanks!
